src/core_modules/capture-core/components/D2Form/FormBuilder/FormBuilder.component.js

```diff
@@ -115,9 +126,15 @@ export class FormBuilder extends React.Component<Props> {
             .reduce(async (passPromise, currentValidator) => {
                 const pass = await passPromise;
                 if (pass === true) {
-                    let result = currentValidator.validator(value, validationContext);
+                    const validatorContext = commitOptions?.errorCode ?
+                        { error: commitOptions.error, errorCode: commitOptions.errorCode } :
+                        validationContext; // For UniqueValidator
```

Comment:
Let's always pass on both the errors and the validationContext so we don't have to potentially change this logic if we are writing new validators. If you pass on errors as the second argument and validatorContext as the third one, I think you only have to change the unqiue validator.